pkg/issuer/acme/dns/util/wait_test.go

```diff
@@ -15,126 +15,292 @@ import (
 	"fmt"
 	"reflect"
 	"sort"
-	"strings"
+	"sync"
+	"sync/atomic"
 	"testing"
 
 	"github.com/miekg/dns"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
-var lookupNameserversTestsOK = []struct {
-	fqdn string
-	nss  []string
-}{
-	{
-		fqdn: "en.wikipedia.org.",
-		nss:  []string{"ns0.wikimedia.org.", "ns1.wikimedia.org.", "ns2.wikimedia.org."},
-	},
-	{
-		fqdn: "www.google.com.",
-		nss:  []string{"ns1.google.com.", "ns2.google.com.", "ns3.google.com.", "ns4.google.com."},
-	},
-	{
-		fqdn: "physics.georgetown.edu.",
-		nss:  []string{"ns.b1ddi.physics.georgetown.edu.", "ns4.georgetown.edu.", "ns5.georgetown.edu.", "ns6.georgetown.edu."},
-	},
-}
-
-var lookupNameserversTestsErr = []struct {
-	fqdn  string
-	error string
-}{
-	// invalid tld
-	{"_null.n0n0.",
-		"Could not determine the zone",
-	},
-}
-
-var findZoneByFqdnTests = []struct {
-	fqdn string
-	zone string
-}{
-	{"mail.google.com.", "google.com."},             // domain is a CNAME
-	{"foo.google.com.", "google.com."},              // domain is a nonexistent subdomain
-	{"example.com.ac.", "ac."},                      // domain is a eTLD
-	{"cross-zone-example.assets.sh.", "assets.sh."}, // domain is a cross-zone CNAME
-}
-
-var checkAuthoritativeNssTestsErr = []struct {
-	fqdn, value string
-	ns          []string
-	error       string
-}{
-	// invalid nameserver
-	{"8.8.8.8.asn.routeviews.org.", "fe01=", []string{"invalid.example.com."},
-		"",
-	},
```

Comment:
This test wasn't testing anything: it was checking that the error returned by `checkAuthoritativeNss` contains `""`, which is always true. I changed this test case to check that `checkAuthoritativeNss` does return the error returned by `DNSQuery` (not much value to it though).